### PR TITLE
Documentation for Bones, Skeleton, and SkinnedMesh

### PR DESCRIPTION
**Pull request by gregtatum**
Let me know if there is anything that needs to be changed.

### CODE REVIEW

**Comment by WestLangley** [comment]
This is really nice.

http://zz85.github.io/mrdoobapproves/ is helpful in identifying formatting issues if you are so inclined -- and reasonably good at auto-correcting them.

**Comment by TatumCreative** [comment]
Great I'll run it through there sometime in the next few days. I'll try and do that for future PRs as well before I send them. This coding style is pretty different than my day-to-day, but I've been trying to make it match.

**Comment by TatumCreative** [comment]
Okie doke, it should pass the validator now. I'll make sure and run my stuff through it going forward.

**Comment by WestLangley** [comment]
Thanks. You inspired me to file https://github.com/zz85/mrdoobapproves/issues/35. : - )

BTW, in my experience, @mrdoob's style leads to fewer coding errors, as the code is more readable.

**Comment by mrdoob** [comment]
Many thanks!

File: docs/api/objects/Skeleton.html [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+		bones.push( shoulder );
+		bones.push( elbow );
+		bones.push( hand );
+		
+		shoulder.position.y = -5;
+		elbow.position.y = 0;
+		hand.position.y = 5;
+		
+		var armSkeleton = THREE.Skeleton( bones );
```
Comment (tdhooper) [inline docs/api/objects/Skeleton.html]: I think this needs a `new`

File: docs/api/objects/SkinnedMesh.html [hunk 2]
```diff
@@ -11,27 +11,60 @@
...
+			var skinIndex = calculateSkinIndex( geometry.vertices, i );
+			var skinWeight = calculateSkinWeight( geometry.vertices, i );
+			
+			// Ease between each bone
+			geometry.skinIndices.push( new THREE.Vector4( skinIndex, skinIndex + 1, 0, 0 ) );
+			geometry.skinWeights.push( new THREE.Vector4( 1 - skinWeight, skinWeight, 0, 0 ) );		
+			
+		}
+		
+		var mesh = THREE.SkinnedMesh( geometry, material );
```
Comment (tdhooper) [inline docs/api/objects/SkinnedMesh.html]: I think this needs a `new`

File: docs/api/objects/Skeleton.html (hunk 1 above)
Comment (mrdoob) [inline docs/api/objects/Skeleton.html]: I think so too :+1:

File: docs/api/objects/SkinnedMesh.html (hunk 2 above)
Comment (mrdoob) [inline docs/api/objects/SkinnedMesh.html]: I think so too :+1:

**Comment by tdhooper** [comment]
Thank you so much for doing this.

I've tried to reproduce your examples here http://codepen.io/tdhooper/pen/KpeQjj but I'm not seeing any changes in the mesh, am I missing something?

**Comment by tdhooper** [comment]
I needed to add `skinning: true` to the material

File: docs/api/objects/SkinnedMesh.html [hunk 3]
```diff
@@ -11,27 +11,60 @@
...
 		<h3>[name]([page:Geometry geometry], [page:Material material], [page:boolean useVertexTexture])</h3>
 		<div>
-                geometry — An instance of [page:Geometry].<br />
-                material — An instance of [page:Material] (optional).<br />
-		useVertexTexture -- Defines wether a vertex texture can be used (optional).
-		</div>
-		<div>
-		This Creates a new instance of skinnedMesh.
+        geometry — An instance of [page:Geometry]. [page:Geometry.skinIndices] and [page:Geometry.skinWeights] should be set.<br />
+        material — An instance of [page:Material] (optional).<br />
```
Comment (tdhooper) [inline docs/api/objects/SkinnedMesh.html]: Could we have a note here that the material needs to have skinning enabled?

**Pull request merged**